src/samples/http/server-sse.c

```diff
@@ -0,0 +1,295 @@
+{
+    int r;
+
+    SOL_BUFFER_DECLARE_STATIC(buf, SOL_INET_ADDR_STRLEN);
+    struct sol_network_link_addr addr;
+    struct sol_http_response response = {
+        SOL_SET_API_VERSION(.api_version = SOL_HTTP_RESPONSE_API_VERSION, )
+        .param = SOL_HTTP_REQUEST_PARAMS_INIT,
+        .response_code = SOL_HTTP_STATUS_OK,
+    };
+
+    r = sol_http_request_get_interface_address(request, &addr);
+    SOL_INT_CHECK(r, < 0, r);
+
+    SOL_NULL_CHECK(sol_network_link_addr_to_str(&addr, &buf), -1);
+
+    sol_buffer_init(&response.content);
+    r = sol_buffer_append_printf(&response.content, HTML_FILE,
+        SOL_STR_SLICE_PRINT(sol_buffer_get_slice(&buf)), port);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_http_server_send_response(request, &response);
+    sol_buffer_fini(&response.content);
+
+    return r;
+}
+
+static void
+startup_server(void)
+{
+    char **argv = sol_argv();
+    int c, opt_idx,  argc = sol_argc();
+    static const struct option opts[] = {
+        { "port", required_argument, NULL, 'p' },
+        { "help", no_argument, NULL, 'h' },
+        { 0, 0, 0, 0 }
+    };
+
+    sol_buffer_init(&value);
+    while ((c = getopt_long(argc, argv, "p:h", opts,
+            &opt_idx)) != -1) {
+        switch (c) {
+        case 'p':
+            port = atoi(optarg);
+            break;
+        case 'h':
+        default:
+            fprintf(stderr,
+                "Usage:\n\t%s [-p <port >]\n\n"
+                "Then everything that is typed will be sent using SSE technique\n"
+                "Test it opening a browser in http//127.0.0.1:<port>\n",
+                argv[0]);
+            sol_quit_with_code(EXIT_SUCCESS);
+            return;
+        }
+    }
+
+    /* always set stdin to non-block before we use sol_fd_add() on it,
+     * otherwise we may block reading and it would impact the main
+     * loop dispatching other events.
+     */
+    if (sol_util_fd_set_flag(STDIN_FILENO, O_NONBLOCK) < 0) {
+        fprintf(stderr, "ERROR: cannot set stdin to non-block.\n");
+        goto err;
+    }
+
+    stdin_watch = sol_fd_add(STDIN_FILENO,
+        SOL_FD_FLAGS_IN | SOL_FD_FLAGS_HUP | SOL_FD_FLAGS_ERR, on_stdin, NULL);
+    if (!stdin_watch) {
+        fprintf(stderr, "ERROR: Failed to watch stdin\n");
+        goto err;
+    }
+
+    server = sol_http_server_new(port);
+    if (!server) {
+        fprintf(stderr, "ERROR: Failed to create the server\n");
+        goto err;
+    }
+
+    if (sol_http_server_register_handler(server, "/",
+        request_cb, NULL) < 0) {
+        fprintf(stderr, "ERROR: Failed to register the handler\n");
+        goto err;
+    }
+
+    if (sol_http_server_register_handler(server, "/events",
+        request_events_cb, NULL) < 0) {
+        fprintf(stderr, "ERROR: Failed to register the handler\n");
+        goto err;
+    }
+
+    printf("HTTP server at port %d.\nDefault reply set to '%.*s'\n",
+        port, SOL_STR_SLICE_PRINT(sol_buffer_get_slice(&value)));
```

Comment:
this "value" here will always be empty since no options or arguments set it.

Please change this to explain to the user he should type something to send to connected servers.